chapter03/Task04.py

```diff
@@ -7,17 +8,26 @@
       '"=" - Если я ответил верно')
 
 number = 50
+a = 0
+b = 101
+con = 0
 
 ans = str()
 while True:
-    ans = str(input('Это число' + str(number)))     #Внести правки по поводу диапозонов, проверки кода, дописать прогу
+    ans = input('Это число ' + str(number) + '?\n')
+    con += 1
     if ans == '=':
         break
     elif ans == '-':
-        number //= 2
+        b = number
+        number = b - (b - a) // 2
+        print(a, b)
     elif ans == '+':
-        number = number+(number//2)
+        a = number
+        number = a + (b - a) // 2
+        print(a, b)
     else:
         continue
 
-print('Я так и знал')
\ No newline at end of file
+print('Я так и знал')
+print('Ты потратил ', con, 'попыток')
```

Comment:
If I understand correctly, that's program spends trials instead of a user.
So maybe "I" instead of "you"?